skimage/transform/_poly2mask.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+import matplotlib
+
+from .. import draw
+
+
+def poly2mask(image_shape, polygon, backend='default'):
+    """Compute a mask from polygon.
+
+    Parameters
+    ----------
+    image_shape : tuple of size 2.
+        The shape of the mask.
+    polygon : array_like.
+        The polygon coordinates of shape (N, 2) where N is
+        the number of points.
+    backend : 'default' or 'matplotlib'.
+        The default implementation uses `skimage.draw`, while the
+        'matplotlib' implementtion uses `maplotlib.path.Path`.
+        While note exactly the same both methods give near-identical results.
+
+    Returns
+    -------
+    mask : 2-D ndarray of type 'bool'.
+        The mask that corresponds to the input polygon.
+    """
+    if backend == 'default':
+        return _poly2mask_skimage(image_shape, polygon)
+    elif backend == 'matplotlib':
+        return _poly2mask_mpl(image_shape, polygon)
+
+    raise ValueError(
+        "Wrong backend selected. Choose either 'default' or 'matplotlib'.")
+
+
+def _poly2mask_skimage(image_shape, polygon):
+    polygon = np.array(polygon)
+    fill_row_coords, fill_col_coords = draw.polygon(
+        polygon[:, 1], polygon[:, 0], image_shape)
```

Comment:
These should not be inverted. I suggest using the same syntax as in my original comments, as it is more self-documenting:

```python
vertex_row_coords, vertex_col_coords = polygon.T
fill_row_coords, fill_col_coords = draw.polygon(vertex_row_coords, vertex_col_coords, image.shape)
```

(accounting for PEP8 of course — hard to do on the github editor)